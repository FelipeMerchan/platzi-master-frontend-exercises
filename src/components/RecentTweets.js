import React from 'react'

import './styles/RecentTweets.css'

function RecentTweets() {
  return (
    <article className="RecentTweets">
      <h2 className="RecentTweets__title">últimos
        <br /> <span>tweets</span>
      </h2>
      <div className="RecentTweet">
        <p>
          <strong><span>5:13PM</span><span>Sep 7</span> desde <span>TweetDeck</span></strong>
        </p>
        <p>
          <span>RT</span><span>@lstwebdesigner</span>: <span>Mira mi último diseño http:/bit.ly/asdfg</span>
        </p>
      </div>
      <div className="RecentTweet">
        <p>
          <strong><span>5:13PM</span><span>Sep 7</span> desde <span>TweetDeck</span></strong>
        </p>
        <p>
          <span>RT</span><span>@lstwebdesigner</span>: <span>Mira mi último diseño http:/bit.ly/asdfg</span>
        </p>
      </div>
    </article>
  )
}

export default RecentTweets