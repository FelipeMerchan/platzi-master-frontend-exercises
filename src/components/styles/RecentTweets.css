.RecentTweets {
  color: #07080d;
}

.RecentTweets__title {
  margin-block-end: var(--space-2);
  font-family: var(--font-4);
  font-size: 40px;
  color: #c5345c;
  line-height: .7;
}

.RecentTweets__title>span {
  color: #66615f;
  padding-inline-start: var(--space-5);
}

.RecentTweet p {
  font-size: 12px;
  margin-block-end: calc(var(--space) / 2);
}

.RecentTweet:nth-of-type(1):after {
  content: '';
  display: inline-block;
  height: 1px;
  width: 80%;
  margin: var(--space-2) 0;
  background-color: #07080d;
}