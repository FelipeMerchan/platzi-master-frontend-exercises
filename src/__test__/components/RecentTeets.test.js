import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import RecentTweets from '../../components/RecentTweets'

describe('Probar el componente <RecentTweets />', () => {
  const recentTweets = shallow(<RecentTweets />)
  test('Render del componente RecentTweets', () => {
    expect(recentTweets.length).toEqual(1)
  })
  test('Render del título', () => {
    expect(recentTweets.exists('.RecentTweets__title')).toBeTruthy()
  })
  test('Render de los dos últimos tweets', () => {
    expect(recentTweets.children()).toHaveLength(3);
  });
})

describe('RecentTweets Snapshot', () => {
  const recentTweets = create(<RecentTweets />)
  test('Comprobar la UI del componente RecentTweets', () => {
    expect(recentTweets.toJSON()).toMatchSnapshot()
  })
})