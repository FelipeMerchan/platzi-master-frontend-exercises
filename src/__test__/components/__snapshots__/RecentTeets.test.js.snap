// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentTweets Snapshot Comprobar la UI del componente RecentTweets 1`] = `
<article
  className="RecentTweets"
>
  <h2
    className="RecentTweets__title"
  >
    últimos
    <br />
     
    <span>
      tweets
    </span>
  </h2>
  <div
    className="RecentTweet"
  >
    <p>
      <strong>
        <span>
          5:13PM
        </span>
        <span>
          Sep 7
        </span>
         desde 
        <span>
          TweetDeck
        </span>
      </strong>
    </p>
    <p>
      <span>
        RT
      </span>
      <span>
        @lstwebdesigner
      </span>
      : 
      <span>
        Mira mi último diseño http:/bit.ly/asdfg
      </span>
    </p>
  </div>
  <div
    className="RecentTweet"
  >
    <p>
      <strong>
        <span>
          5:13PM
        </span>
        <span>
          Sep 7
        </span>
         desde 
        <span>
          TweetDeck
        </span>
      </strong>
    </p>
    <p>
      <span>
        RT
      </span>
      <span>
        @lstwebdesigner
      </span>
      : 
      <span>
        Mira mi último diseño http:/bit.ly/asdfg
      </span>
    </p>
  </div>
</article>
`;
